merlin/main.py

```diff
@@ -929,6 +932,99 @@ def generate_diagnostic_parsers(subparsers: ArgumentParser) -> None:
         "Example: '--vars LEARN=path/to/new_learn.py EPOCHS=3'",
     )
 
+    # merlin detailed-status
+    detailed_status: ArgumentParser = subparsers.add_parser(
+        "detailed-status",
+        help="Display a task-by-task status of a study.",
+    )
+    detailed_status.set_defaults(func=query_status, detailed=True)
+    detailed_status.add_argument(
+        "spec_or_workspace", type=str, help="Path to a Merlin YAML spec file or a launched Merlin study"
+    )
+    detailed_status.add_argument(
+        "--dump", type=str, help="Dump the status to a file. Provide the filename (must be .csv or .json).", default=None
+    )
+    detailed_status.add_argument(
+        "--task_server",
+        type=str,
+        default="celery",
+        help="Task server type.\
+                            Default: %(default)s",
+    )
+    detailed_status.add_argument(
+        "--vars",
+        action="store",
+        dest="variables",
+        type=str,
+        nargs="+",
+        default=None,
+        help="Specify desired Merlin variable values to override those found in the specification. Space-delimited. "
```

Comment:
What does it mean to override spec variables for the status command specifically?  Isn't that already captured in the study workspace and not to be changed further?

Reply:
Yes but with the status/detailed-status command I implemented you can pass in a spec file as well as an output directory (e.g. `merlin status my_study.yaml` or `merlin status my_study_20231010-140705/`). When a spec file is passed in, Merlin will do some behind the scenes work and look for any potential studies that exist w.r.t. that spec file. It will then ask the user which one they want to view the status of:
![status-potential-studies](https://github.com/LLNL/merlin/assets/49216024/57042c4e-e58c-4ebc-afbd-fc71eed9a928)

It does all of this by looking in the `OUTPUT_PATH` for directories that match the name of the study and have a time stamp. So, if a user modified the `OUTPUT_PATH` variable at the CLI for their run then they would need to modify the `OUTPUT_PATH` again if they decided to pass a spec in as input to the status command or else Merlin would be unable to find the study they're looking for. If they pass in an output directory rather than a spec file then this no longer matters, but I was just trying to think of any possible scenario users might run into with this format. Does this make sense?

Reply:
Thinking this through a bit more, I'm pretty sure `OUTPUT_PATH` would be the only variable that would need to be manipulated at the CLI for `status` and `detailed-status`. To make this more clear, maybe I should just add a `--output-path` option and remove the `--vars` option completely?